src/payment/strategies/zip/zip-payment-strategy.ts

```diff
@@ -56,14 +58,15 @@ export default class ZipPaymentStrategy implements PaymentStrategy {
             throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
         }
 
-        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
-
-        const { useStoreCredit } = payload;
+        const { isStoreCreditApplied: useStoreCredit } = this._store.getState().checkout.getCheckoutOrThrow();
 
         if (useStoreCredit !== undefined) {
             await this._store.dispatch(this._storeCreditActionCreator.applyStoreCredit(useStoreCredit));
         }
 
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+        await this._store.dispatch(this._remoteCheckoutActionCreator.initializePayment(payment.methodId, { useStoreCredit }));
```

Comment:
I don't think you need to call remoteCheckoutActionCreator. The store credit was probably not working because the order was being created first. Store credit needs to happen first.

Reply:
It will not work properly if if the remoteCheckoutActionCreator is not called, when store  credit is applied it unsets a session variable in the backend which is validated in the backend when the payment method is loaded (to get the client token) in line 70, without this variable the token is generated without the store credit applied, the remoteCheckoutActionCreator sets the variable and allows the token to be generated with the store credit applied.

Reply:
Thanks for explaining. Part of the idea behind the new StoreCreditActionCreator was to remove the need of using `remoteCheckoutActionCreator` for store credit purposes, but I guess there is more work required to be done in the backend to fix it.